src/components/userProfile/sagas.js

```diff
@@ -0,0 +1,39 @@
+import { effects } from 'redux-saga';
+
+import App from '../app';
+import { fetchUserData } from './actions';
+import { REQUEST_FETCH_USER_DATA } from './actionTypes';
+
+const { selectors: { getUserData } } = App;
+
+function* findUserData(){
+    try {
+        const user = yield effects.select(getUserData);
+        // eslint-disable-next-line no-console
+        console.log(user);
+        if (user) {
+            yield effects.put(fetchUserData(user));
+        } else {
+            // eslint-disable-next-line no-console
+            console.log("User data not fetched");
+        }
+    } catch(error){
+        // eslint-disable-next-line no-console
+        console.log(error);
+    }
+}
```

Comment:
I am glad that you were able to figure this saga flow out, however, this is not needed given that the user information should already be accessible before you show the profile.

Reply:
in the app component there is a selector called `getUserData` use that to access user data not make another request to the server for data you already have